database.go

```diff
@@ -136,6 +136,7 @@ func (db *Database) LockProtectedEntries() error {
 	if err != nil {
 		return err
 	}
+	db.cleanupBinaries()
```

Comment:
How does the binary cleanup relate to locking protected entries?

As far as I can tell none of the binary data is covered here

Reply:
You wouldn't necessarily have to unlock protected entries to change the attached files.

That is also the whole reason why it is a separate operation to unlock and lock protected entries rather than always having it done automatically.

Therefore the cleanup call rather belongs into the encoder